glue/core/data_factories/npy.py

```diff
@@ -74,9 +89,8 @@ def npz_reader(filename, format='auto', auto_merge=False, **kwargs):
         d = Data(label=groupname)
         arr = npy_data[groupname]
 
-        if not hasattr(arr.dtype, 'names'):
-            raise ValueError("Numpy save file loading currently only supports structured"
-                             " arrays, e.g., with specified names.")
+        if isinstance(arr.dtype.names, type(None)):
+            arr = from_unstructured_array(arr)
```

Comment:
Same comment as for the ``npy`` reader, just add the single component manually and avoid making a recarray.